config_utilities/include/config_utilities/config.h

```diff
@@ -45,6 +45,8 @@
 #include "config_utilities/internal/namespacing.h"
 #include "config_utilities/internal/visitor.h"
 #include "config_utilities/traits.h"
+#include "config_utilities/types/enum.h"
```

Comment:
These two includes are probably not needed here (with the changes suggested below)